psydac/api/grid.py

```diff
@@ -157,6 +157,9 @@ def __init__( self, V, nderiv , trial=False, grid=None, ext=None):
                 points = grid.points[axis]
                 boundary_basis = basis_ders_on_quad_grid(
                         space.knots, space.degree, points, nderiv, space.basis)
+                if space.basis == 'M':
+                    import numpy as np
+                    boundary_basis = np.zeros_like(boundary_basis)
```

Comment:
some components of the space are zero when we are in the boundary, and this ensures it.
this is something we didn't agree on yet, so I added the hack outside of the  `basis_ders_on_quad_grid`

Reply:
I see, then please do these two things:

1. Add one (or more) line(s) of comment before this three lines of code, where you explain with a bit of detail what you are doing and why;
2. Instead of using `zeros_like` from `numpy`, you can just set the array to zero:
```python
if space.basis == 'M':
    boundary_basis[...] = 0.0

Reply:
By the way, `basis_ders_on_quad_grid` computes the values of the 1D basis functions, so this "hack" is definitely not related to it.

And I am still not clear why a component of the space should be zero on the boundary, unless we are using homogeneous Dirichlet boundary conditions. Can you give us an example?

Reply:
if you take a unit square domain, the `Hcurl` space on the boundary `x = 1` is equal to `(0, N(x=1)*M)`

Reply:
But as Yaman was pointing, this is only true for a function in H_0(curl), not for a general H(curl) function.

Reply:
By the way, the docstring of the `BasisValues` class should be more detailed. 

In particular, it should specify on which grid the values are computed,
and also probably indicate that the derivatives of the basis values are computed there (as it seems). 

If the parameter 'grid' is only for the boundaries, it should probably be called boundary_grid.

Reply:
yes,it's when we have homogenous boundary conditions

Reply:
so in this case the values on the boundary should be zero. If there are not, it's because of an error somewhere else. This hack may just hide that.

Reply:
yes we set them here, there is no other place

Reply:
I have deleted it, untill we properly implement homogenous boundary conditions